docs/dxp/latest/en/process-automation/forms/developer-guide/writing-a-custom-form-field-type.md

```diff
@@ -81,57 +82,65 @@ To see how custom form fields work, deploy an example and then add some form dat
+- `ddm.form.field.type.description`: provide the language key for the description text. Make sure the translated value is defined in the `Language.properties` file.
+- `ddm.form.field.type.display.order`: set an integer value to determine where the field is displayed in the Form Builder sidebar. If there's a tie, the fields are sorted alphabetically.
+- `ddm.form.field.type.group`:
+- `ddm.form.field.type.icon`: decide which icon type to use for your field. Choose from blah blah
+- `ddm.form.field.type.label`: provide the language key for the label text. Make sure the translated value is defined in the `Language.properties` file.
+- `ddm.form.field.type.name`: provide the language key for the field name. Make sure the translated value is defined in the `Language.properties` file.
 
-1. `className="ddm-field-slider form-control slider"` blah blah blah
-1. `id="myRange"` blah blah blah
-1. `type="range"` defines the type of data that will be passed as a value to the field. The `range` type means that the fields accepts an integer.
-1. `max={100}` specifies that the maximum value for the field will be `100`.
-1. `min={1}` sets the minimum accepted value at `1`.
+The `getModuleName` method passes the `Slider.es.js` file path to the `NPMResolver` service. Some of the path definition is accomplished in the `package.json` file (see the `name` declaration and the `source-maps` defined in the `scripts` section).
 
-Below you can find steps showing you how to make the max and min properties configurable by the form builder.
+```{literalinclude} ./writing-a-custom-form-field-type/resources/liferay-c2p9.zip/c2p9-impl/src/main/java/com/acme/c2p9/internal/dynamic/data/mapping/form/field/type/C2P9DDMFormFieldType.java
+   :dedent: 1
+   :language: java
+   :lines: 22-26,38-39
+```
 
-The rest of the properties () are defined by the Main block and its function. When the exported `Main` variable is called by the Forms Framework, .
+The `getName` method returns the language key defining the name that will appear in the UI: `c2p9-slider` in this case, which is translated to `C2P9 Slider` in the `c2p9-impl/src/main/resources/content/Language.properties` file.
 
-What are the `otherProps`: Required, Help Text, Show Label, Repeatable?
+```{literalinclude} ./writing-a-custom-form-field-type/resources/liferay-c2p9.zip/c2p9-impl/src/main/java/com/acme/c2p9/internal/dynamic/data/mapping/form/field/type/C2P9DDMFormFieldType.java
+   :dedent: 1
+   :language: java
+   :lines: 28-31
+```
 
-```{literalinclude} ./writing-a-custom-form-field-type/resources/liferay-c2p9.zip/c2p9-impl/src/main/resources/META-INF/resources/C2P9/Slider.es.js
-   :dedent: 0
-   :language: js
-   :lines: 5-17
+The `isCustomDDMFormFieldType` method must return `true` for all form fields not shipped out of the box with Liferay.
```

Comment:
I'm not sure if this is relevant information to the dev, but the method `isCustomDDMFormFieldType` is only used to determine if we need to resolve the NPM module name for this field type. Meaning, if its return is `false`, then we will call the `NPMResolver.resolveModuleName` for the return of the `getModuleName`. Finally, if you're already calling the NPM Resolver inside the `getModuleName`, then `isCustomDDMFormFieldType` can return `true`.

Reply:
Do we expect that all custom form fields will set this to true?

Reply:
We don't. This is just used for what I've mentioned. I believe this is just bad naming.

Reply:
> The `isCustomDDMFormFieldType` is used internally. Return `true` if you're returning the result of `NPMResolver.resolveModuleName()` in the `getModuleName` method.

How's that? I didn't explain what happens if you return `false` because it doesn't seem like something the reader really needs to know. Let me know if you disagree.

Reply:
Looks good to me, @rbohl! 👍🏻